core/src/main/java/org/aerogear/mobile/core/configuration/MobileCoreJsonParser.java

```diff
@@ -49,7 +49,7 @@ private String readJsonStream(final InputStream jsonStream) throws IOException {
         Map<String, ServiceConfiguration> serviceConfigs = new HashMap<>();
```

Comment:
You already know how many elements you are going to put inside the map (length).
Better initialize this to:
```java
Map<String, ServiceConfiguration> serviceConfigs = new HashMap<>(length);
```

to avoid allocating too much/few memory